addons/event_sale/models/sale_order.py

```diff
@@ -83,6 +87,13 @@ def _onchange_event_ticket_id(self):
         # we call this to force update the default name
         self.product_id_change()
 
+    def unlink(self):
+        self._delete_associated_registrations()
+        super(SaleOrderLine, self).unlink()
+
+    def _delete_associated_registrations(self):
+        self.mapped('event_id.registration_ids').filtered(lambda r: r.sale_order_line_id.id in self.ids).unlink()
```

Comment:
Maybe ``self.env['event.registration'].search([('sale_order_line_id', 'in', self.ids)]).unlink`` ?